src/views/search/types.ts

```diff
@@ -173,7 +188,97 @@ export const UiText = (mode: FcsMode): string => {
   return codeToUiText[mode];
 };
 
+export type NoneReranker = { type: "none" };
+
+export type CustomerSpecificReranker = {
+  type: "customer_reranker";
+  reranker_id: string;
+};
+
+export type MmrReranker = {
+  type: "mmr";
+  diversity_bias: number;
+};
+
+export type SearchConfiguration = {
+  corpora: {
+    corpus_key: string;
+    metadata_filter?: string;
+    lexical_interpolation?: number;
+    custom_dimensions?: Record<string, number>;
+    semantics?: "default" | "query" | "response";
+  }[];
+  offset: number;
+  limit?: number;
+  context_configuration?: {
+    characters_before?: number;
+    characters_after?: number;
+    sentences_before?: number;
+    sentences_after?: number;
+    start_tag?: string;
+    end_tag?: string;
+  };
+  reranker?: NoneReranker | CustomerSpecificReranker | MmrReranker;
+};
+
+export type NoneCitations = {
+  style: "none";
+};
+
+export type NumericCitations = {
+  style: "numeric";
+};
+
+export type HtmlCitations = {
+  style: "html";
+  url_pattern: string;
+  text_pattern: string;
+};
+
+export type MarkdownCitations = {
+  style: "markdown";
+  url_pattern: string;
+  text_pattern: string;
+};
 
+export type GenerationConfiguration = {
+  prompt_name?: string;
+  max_used_search_results?: number;
+  prompt_text?: string;
+  max_response_characters?: number;
+  response_language?: SummaryLanguage;
+  model_parameters?: {
+    max_tokens: number;
+    temperature: number;
+    frequency_penalty: number;
+    presence_penalty: number;
+  };
+  citations?: NoneCitations | NumericCitations | HtmlCitations | MarkdownCitations;
+  enable_factual_consistency_score?: boolean;
+};
+
+export type QueryBody = {
+  query: string;
+  search: SearchConfiguration;
+  stream_response?: boolean;
+  generation?: GenerationConfiguration;
+};
+
+export type QueryRequestHeaders = {
+  ["customer-id"]: string;
+  ["Content-Type"]: string;
+  ["x-api-key"]?: string;
+  ["Authorization"]?: string;
+};
+
+export type ApiV2SearchResponse = {
+  search_results: SearchResult[];
+  factual_consistency_score: number;
+  response_language: string;
+  summary: string;
+  document: SearchResponseDoc[];
+  response: SearchResponseResult[];
+}
```

Comment:
maybe reorder the reranker, so they are in the right order (normal, mmr, slingshot)?